server/app/auth.go

```diff
@@ -57,18 +113,45 @@ func handleLogin(ds model.DataStore, username string, password string, w http.Re
 		_ = rest.RespondWithError(w, http.StatusInternalServerError, "Unknown error authenticating user. Please try again")
 		return
 	}
+	payload := buildPayload(user, tokenString)
+	_ = rest.RespondWithJSON(w, http.StatusOK, payload)
+}
+
+func buildPayload(user *model.User, tokenString string) map[string]interface{} {
 	payload := map[string]interface{}{
-		"message":  "User '" + username + "' authenticated successfully",
+		"message":  "User '" + user.UserName + "' authenticated successfully",
 		"token":    tokenString,
 		"id":       user.ID,
 		"name":     user.Name,
-		"username": username,
+		"username": user.UserName,
 		"isAdmin":  user.IsAdmin,
 	}
 	if conf.Server.EnableGravatar && user.Email != "" {
 		payload["avatar"] = gravatar.Url(user.Email, 50)
 	}
-	_ = rest.RespondWithJSON(w, http.StatusOK, payload)
+	return payload
+}
+
+func validateIPAgainstList(ip string, comaSeparatedList string) bool {
+	if comaSeparatedList == "" || ip == "" {
+		return false
+	}
+	ip = strings.Split(ip, ":")[0]
+	cidrs := strings.Split(comaSeparatedList, ",")
+	testedIP, _, err := net.ParseCIDR(fmt.Sprintf("%s/32", ip))
```

Comment:
I'll fix IPv6 support, I completely forgot about it. Regarding Host IPs, I'm afraid I don't understand the question. 

My understanding is, that `r.RemoteAddr` holds the IP of the connecting client (i.e. "raw" IP, not the one passed via `X-*` headers). So in this case it should be a proxy IP. We can check this proxy IP against whitelist to ensure, that no other party is trying to tamper the headers and log in bypassing authentication, by connecting directly to the app.

Reply:
By "Host" I mean single address and not a network address (with /size). Rephrasing my question: Why we have to specify the size of the network in each item of `ReverseProxyWhitelist` (like `192.168.1.1/32`)? Why not just a list of IPs like `192.168.1.1,fdf8:f53b:82e4::53`

Reply:
Because sometimes (especially in docker containers) user may not be sure, what IP his proxy has. He only knows that it's in some specific subnet. Also, most of the projects (like gitea or grafana) which implements this feature, use CIDRs, so I believe it's somehow a standard.